index.bs

```diff
@@ -1477,102 +1501,122 @@ RemoteValue = {
   WindowProxyRemoteValue //
 }
 
+InternalId = String;
+
 ListRemoteValue = [*RemoteValue];
 
 MappingRemoteValue = [*[(RemoteValue / text), RemoteValue]];
 
 SymbolRemoteValue = {
-   type: "symbol",
-   objectId: ObjectId,
+  type: "symbol",
+  ?handle: Handle,
+  ?internalId: InternalId,
```

Comment:
I guess having string provides more flexibility. When we switch to a general weakref, I imagine we want to keep some additional internal info in the ID like it's done in CDP's `objectId`. On the implementation side you still can keep it as number, and make `.toString()` to save some bites.

Reply:
A string that isn't just random but encodes additional information sounds like a problem, not a goal. If some implementations are encoding data into the element ids then clients are likely to start trying to parse the ids instead of using  them as opaque identifiers. Then we either end up having to specify the id format (which isn't designed to be compatible with multiple implementations and so might be difficult) or we end up with clients that are tied to specific implementations.

Reply:
My suggestion is to keep string IDs to provide flexibility on the implementation side.

Reply:
Honestly if this is just an internal id that's fully set during serialization I wonder if we should just specify it as a counter i.e. the first object involved in a cycle gets internalId 0, the next one gets internal id 1, etc. That minimises the chance that people expect the id to work outside the specific serialization.

I'm not exactly sure what flexibility you're looking for on the implementation side, but if the plan is to expose the CDP objectId directly that would seem to be an interop hazard since a client might come to depend on that id being usable in other parts of the API. Generally I think we should either a) make the id here purely something that the local end (client) can use to deserialize the specific object graph, and useless outside of that or b) revisit the possibility of just using `objectId` and having a future upgrade path to `objectId` references that act as global identifiers for the object but don't express ownership.

Reply:
I made an `internalId` a unique per serialization number. I guess it prevents uses from relying on them in any other then an expected way.